server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -111,6 +120,152 @@ private Optional<String> validateDateOfBirth(String dob) {
+    }
+    return form;
+  }
+
+  private Form<EditTiClientInfoForm> validateEmailAddress(
+      Form<EditTiClientInfoForm> form, AccountModel currentAccount) {
+    String newEmail = form.get().getEmailAddress();
+    // email addresses not a requirement for TI Client
+    if (Strings.isNullOrEmpty(newEmail)) {
+      return form;
+    }
+    if (hasEmailChanged(newEmail, currentAccount)
+        && accountRepository.lookupAccountByEmail(newEmail).isPresent()) {
+      return form.withError(
+          FORM_FIELD_NAME_EMAIL_ADDRESS,
+          "Email address already in use. Cannot update applicant if an account already"
+              + " exists.");
+    }
+    return form;
+  }
+
+  /**
+   * This function updates the client Information after validating the form fields
+   *
+   * @param tiGroup - the TIGroup who manages the account whose info needs to be updated.
+   * @param accountId - the account Id of the applicant whose info should be updated
+   * @param form - this contains all the fields like dob, phoneNumber, emailAddress, name and
+   *     tiNotes.
+   * @return form - the form object is always returned. If the form contains error, the controller
+   *     will handle the field messages. If the account is not found for the given AccountId, a
+   *     runtime exception is raised.
+   */
+  public Form<EditTiClientInfoForm> updateClientInfo(
+      Form<EditTiClientInfoForm> form, TrustedIntermediaryGroupModel tiGroup, Long accountId)
+      throws ApplicantNotFoundException {
+    // validate functions return the form w/ validation errors if applicable
+    form = validateFirstNameForEditClient(form);
+    form = validateLastNameForEditClient(form);
+    form = validatePhoneNumber(form);
+    form = validateDateOfBirth(form);
+    if (form.hasErrors()) {
+      return form;
+    }
+    Optional<AccountModel> accountMaybe =
+        tiGroup.getManagedAccounts().stream()
+            .filter(account -> account.id.equals(accountId))
+            .findAny();
+    if (accountMaybe.isEmpty() || accountMaybe.get().newestApplicant().isEmpty()) {
+      throw new ApplicantNotFoundException(accountId);
+    }
+    form = validateEmailAddress(form, accountMaybe.get());
+    if (form.hasErrors()) {
+      return form;
+    }
+    ApplicantModel applicant = accountMaybe.get().newestApplicant().get();
+    ApplicantData applicantData = applicant.getApplicantData();
+
+    // name update
+    String firstName = form.get().getFirstName();
+    String middleName = form.get().getMiddleName();
+    String lastName = form.get().getLastName();
+    if (isNameChanged(firstName, middleName, lastName, applicantData)) {
+      accountRepository.updateClientName(firstName, middleName, lastName, applicant);
+    }
+    // DOB update
+    String newDob = form.get().getDob();
+    LocalDate newDobDate = LocalDate.parse(newDob, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    if (!applicantData.getDateOfBirth().get().equals(newDobDate)) {
```

Comment:
@swatkat1 You need to add a check for `Optional.Empty()` when calling `applicantData.getDateOfBirth()` to get past the `NoSuchElementException`. You should also do that to the other calls off of applicantData as well.

Reply:
@gwendolyngoetz  - I am seeing this error- `[error] Test services.ti.TrustedIntermediaryServiceTest.editTiClientInfo_AllPass_NameEmailUpdate failed: org.junit.ComparisonFailure: expected:<["4259879090"]> but was:<[Optional.empty]>, took 0.203 sec
[error]     at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[error]     at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[error]     at services.ti.TrustedIntermediaryServiceTest.editTiClientInfo_AllPass_NameEmailUpdate(TrustedIntermediaryServiceTest.java:398)
` It is failing only for Phone number which is wellknownpath added in this PR

Reply:
with transaction support it works.